packages/webpack/src/utils/module-federation/share.ts

```diff
@@ -66,14 +76,59 @@ export function shareWorkspaceLibraries(
         (aliases, library) => ({ ...aliases, [library.name]: library.path }),
         {}
       ),
-    getLibraries: (eager?: boolean): Record<string, SharedLibraryConfig> =>
-      pathMappings.reduce(
-        (libraries, library) => ({
+    getLibraries: (eager?: boolean): Record<string, SharedLibraryConfig> => {
+      let pkgJson = null;
+      if (
+        project?.root &&
+        existsSync(
+          joinPathFragments(workspaceRoot, project.root, 'package.json')
+        )
+      ) {
+        pkgJson = readJsonFile(
+          joinPathFragments(workspaceRoot, project.root, 'package.json')
+        );
+      }
+      return pathMappings.reduce((libraries, library) => {
+        // Check to see if the library version ranged is declared in the root package.json
+        let version =
+          pkgJson?.dependencies?.[library.name] ??
+          pkgJson?.devDependencies?.[library.name] ??
+          null;
+        if (!version) {
+          const workspaceLib = workspaceLibs.find(
+            (lib) => lib.importKey === library.name
+          );
+
+          const workspacePkgJsonPath = join(workspaceLib.root, 'package.json');
+          if (existsSync(workspacePkgJsonPath)) {
+            const workspacePackageJson: PackageJson =
+              readJsonFile(workspacePkgJsonPath);
+
+            if (!workspacePackageJson) {
+              logger.warn(
+                `Could not find a version for "${library.name}" in the root "package.json"`
```

Comment:
I don't think we should print a warning. If the user wants to use singletons without requiredVersion then I think they should be able to. Warnings should let the user know how to remedy the situation, but in this case I think it's a legit use-case.

Reply:
This should not be there anymore will be removed